client/ayon_applications/plugins/launcher_actions/debug_shell.py

```diff
@@ -0,0 +1,175 @@
+import os
+from typing import Optional
+
+from qtpy import QtWidgets, QtGui, QtCore
+
+from ayon_applications import (
+    Application,
+    ApplicationManager,
+    APPLICATIONS_ADDON_ROOT
+)
+from ayon_applications.utils import get_app_environments_for_context
+from ayon_core.pipeline import LauncherAction
+from ayon_core.style import load_stylesheet
+from ayon_core.tools.utils.lib import get_qt_icon
+
+
+def get_application_qt_icon(application: Application) -> Optional[QtGui.QIcon]:
+    """Return QtGui.QIcon for an Application
+
+    Note: This forces the icons to be search in `ayon_applications/icons`
+        folder on client side and mimics what Ayon applications addon's
+        `get_app_icon_path` method does.
```

Comment:
If that's the case then please use `get_app_icon_path` from addon, or move the function to `utils.py`, because right now this is different logic from what is in addon method.